frame/staking/src/lib.rs

```diff
@@ -1597,53 +1692,99 @@ decl_module! {
 		/// Force there to be a new era at the end of the next session. After this, it will be
 		/// reset to normal (non-forced) behaviour.
 		///
+		/// The dispatch origin must be Root.
+		///
 		/// # <weight>
 		/// - No arguments.
+		/// - Base Weight: 1.959 µs
+		/// - Write ForceEra
 		/// # </weight>
-		#[weight = 5_000_000]
+		#[weight = 2 * WEIGHT_PER_MICROS + T::DbWeight::get().writes(1)]
 		fn force_new_era(origin) {
 			ensure_root(origin)?;
 			ForceEra::put(Forcing::ForceNew);
 		}
 
 		/// Set the validators who cannot be slashed (if any).
-		#[weight = 5_000_000]
+		///
+		/// The dispatch origin must be Root.
+		///
+		/// # <weight>
+		/// - O(V)
+		/// - Base Weight: 2.208 + .006 * V µs
+		/// - Write: Invulnerables
+		/// # </weight>
+		#[weight = T::DbWeight::get().writes(1)
+			.saturating_add(2 * WEIGHT_PER_MICROS)
+			.saturating_add((6 * WEIGHT_PER_NANOS).saturating_mul(validators.len() as Weight))
+		]
 		fn set_invulnerables(origin, validators: Vec<T::AccountId>) {
 			ensure_root(origin)?;
 			<Invulnerables<T>>::put(validators);
 		}
 
 		/// Force a current staker to become completely unstaked, immediately.
-		#[weight = 0]
-		fn force_unstake(origin, stash: T::AccountId) {
+		///
+		/// The dispatch origin must be Root.
+		///
+		/// # <weight>
+		/// O(S) where S is the number of slashing spans to be removed
+		/// Base Weight: 53.07 + 2.365 * S µs
+		/// Reads: Bonded, Slashing Spans, Account, Locks
+		/// Writes: Bonded, Slashing Spans (if S > 0), Ledger, Payee, Validators, Nominators, Account, Locks
+		/// Writes Each: SpanSlash * S
+		/// # </weight>
+		#[weight = T::DbWeight::get().reads_writes(4, 7)
+			.saturating_add(48 * WEIGHT_PER_MICROS)
+			.saturating_add(
+				WEIGHT_PER_MICROS.saturating_mul(2).saturating_mul(Weight::from(*num_slashing_spans))
+			)
+			.saturating_add(T::DbWeight::get().writes(Weight::from(*num_slashing_spans)))
+			// if slashing spans is non-zero, add 1 more write
+			.saturating_add(T::DbWeight::get().writes(Weight::from(*num_slashing_spans).min(1)))
```

Comment:
(sidenote: actually u64 implements `From<bool>`)
```suggestion
			.saturating_add(T::DbWeight::get().writes(Weight::from(*num_slashing_spans > 0)))
```

Reply:
💡